internal/controller/pkg/revision/hook.go

```diff
@@ -116,6 +118,9 @@ func (h *ProviderHooks) Pre(ctx context.Context, pkg runtime.Object, pr v1.Packa
 	// NOTE(hasheddan): we avoid fetching pull secrets and controller config as
 	// they aren't needed to delete Deployment, ServiceAccount, and Service.
 	s, d, svc, secSer, secCli := buildProviderDeployment(pkgProvider, pr, nil, h.namespace, []corev1.LocalObjectReference{})
+	if secSer == nil || secCli == nil {
```

Comment:
do we really want to error out here? or would it be sufficient to skip the deletion of the missing secrets?

Reply:
btw if we finish up #4305 we won't have to delete them anymore, right? similar to what we do for functions.